### PR TITLE
Ast endpoints

### PR DESCRIPTION
**Pull request by c-brenn**
* `/api/pml` returns the AST as part of the repsonse - see readme

* `/api/ast` takes an AST and returns the corresponding PML as an attachment

### CODE REVIEW

**Comment by 22a** [comment]
added a pml download button

it looks and works like this:

**Comment by 22a** [comment]
The time at which we call [these](https://github.com/tom-and-the-toothfairies/pathways/pull/146/files#diff-af326ae966b05ea64b817eac7636661eR83) is up for debate, the functions were designed in such a way so as to facilitate multiple ast revisions (necessary if PML-TX features are purchased)

File: panacea/web/static/js/app.js [hunk 1]
```diff
@@ -48,7 +48,30 @@ async function handleDrugsResponse(response) {
...
-async function handleUrisResponse(response) {
+const triggerDownload = (fileName, fileContents) => {
+  const element = document.createElement('a');
+  element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(fileContents));
+  element.setAttribute('download', fileName);
+  element.style.display = 'none';
+  document.body.appendChild(element);
+  element.click();
+  document.body.removeChild(element);
+};
```
Comment (houli) [inline panacea/web/static/js/app.js]: Instead of this stuff we can use `Blob` api I believe

File: panacea/web/static/js/app.js [hunk 2]
```diff
@@ -48,7 +48,30 @@ async function handleDrugsResponse(response) {
...
+
+const prepareDownloadButton = ast => {
+  const pmlDownloadButton = document.getElementById('pml-download-button');
+  pmlDownloadButton.onclick = async function(){
```
Comment (houli) [inline panacea/web/static/js/app.js]: addEventListener

File: panacea/web/static/css/app.css [hunk 3]
```diff
@@ -34,3 +34,7 @@
   height: 128px;
   margin: 40px;
 }
+
+.padded-bottom {
```
Comment (houli) [inline panacea/web/static/css/app.css:38]: Instead of this padded class will using bootstrap `row` bits get us the right padding we want?

**Review by houli — CHANGES_REQUESTED** [review]

File: panacea/web/static/js/request.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const apiAccessToken = document.getElementById('api-access-token').content;
+export const apiAccessToken = document.getElementById('api-access-token').content;
```
Comment (houli) [inline panacea/web/static/js/request.js]: Don't need an export any more if I'm not mistaken

**Review by houli — APPROVED** [review]

**Pull request merged**